.github/workflows/merge-up.yml

```diff
@@ -0,0 +1,34 @@
+name: Merge up
+
+on:
+  push:
+    branches:
+      - "v[0-9]+.[0-9]+"
+
+permissions:
+  contents: write
+  pull-requests: write
+
+env:
+  GH_TOKEN: ${{ github.token }}
+
+jobs:
+  merge-up:
+    name: Create merge up pull request
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v4
+        with:
+          # fetch-depth 0 is required to fetch all branches, not just the branch being built
+          fetch-depth: 0
+
+      - name: Create pull request
+        id: create-pull-request
+        uses: alcaeus/automatic-merge-up-action@main
+        with:
+          ref: ${{ github.ref_name }}
+          branchNamePattern: 'v<major>.<minor>'
+          fallbackBranch: 'master'
```

Comment:
This ensures that if no newer versioned branch name is found, a pull request to `master` is created. The list of branches checked when pushing to v1.17 would thus be:
* v1.18
* v2.0
* master

Reply:
Noted that this is the only difference from https://github.com/mongodb/laravel-mongodb/pull/2748